### ISSUE TITLE
Incorrect mouse position

### ISSUE THREAD

**Issue by DivineGod** [alacritty/alacritty#3191]
Using the mouse to select text is off when on Retina screen.

Bug introduced in https://github.com/alacritty/alacritty/commit/c34ec12c309695e4c14d8e50b5f3f54198f70775

### System

OS: macOS
Version: master

Reproduction:

Open Alacritty on retina screen
Try to select text
Selection is off from actual cursor position

**Comment by DivineGod** [comment]
start-up log:

[Alacritty-91939.log](https://github.com/alacritty/alacritty/files/4050322/Alacritty-91939.log)

**Comment by kchibisov** [comment]
opened upstream issue https://github.com/rust-windowing/winit/issues/1389

**Comment by DivineGod** [comment]
Confirmed by adding the following to top-level Cargo.toml

```toml
[patch.crates-io]
winit = { git = 'https://github.com/rust-windowing/winit' }
```

that winit `master` fixes the issue.

**Comment by ddevault** [comment]
Can reproduce on Linux/Wayland as well.

**Comment by kchibisov** [comment]
@ddevault are you using some master version of alacritty? it shouldn't be a problem on v0.4.1.

**Comment by ddevault** [comment]
Aye I'm using master, I switched to the stable release for now.

**Comment by tinmarino** [comment]
PR #3307 solved it here (Ubuntu 20.10 / Tmux 3.1b). The Shift left click was one character left and one above the mouse position.
Now it is perfect!
Thanks devs!
